`error: 'syscall' is deprecated` on macOS (1.20.2)

macOS 15.4.1 24E263 arm64

```
building '/nix/store/4lb0mg0y4is2nvbj9xbqvfa227jkp7hj-libiscsi-1.20.2.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/nrbj29f92zk6f6z2df01hk5wc43xzp5r-source
source root is source
Running phase: patchPhase
Running phase: autoreconfPhase
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
aclocal: warning: couldn't open directory 'm4': No such file or directory
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I m4
autoreconf: running: /nix/store/al2a0kqslc7s5wqcq5rybv1qnbi0kxnj-autoconf-2.72/bin/autoconf --force
autoreconf: running: /nix/store/al2a0kqslc7s5wqcq5rybv1qnbi0kxnj-autoconf-2.72/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:8: installing './ar-lib'
configure.ac:8: installing './compile'
configure.ac:9: installing './config.guess'
configure.ac:9: installing './config.sub'
configure.ac:6: installing './install-sh'
configure.ac:6: installing './missing'
examples/Makefile.am: installing './depcomp'
autoreconf: './install-sh' is updated
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
fixing libtool script ./ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/2dxm3wzlk7h4scfh1qfwmgq7v9kg7qly-libiscsi-1.20.2
checking for a BSD-compatible install... /nix/store/7snvxb9g7cb42vrma4almw8f5wh409rz-coreutils-9.7/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/7snvxb9g7cb42vrma4almw8f5wh409rz-coreutils-9.7/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether clang accepts -g... yes
checking for clang option to enable C11 features... none needed
checking whether clang understands -c and -o together... yes
checking dependency style of clang... none
checking the archiver (ar) interface... ar
checking build system type... aarch64-apple-darwin24.4.0
checking host system type... aarch64-apple-darwin24.4.0
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/lx4icjsdgqqs8j5nqrkfq72jbcjn3zgz-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/phssz9lgny3mwx26w3z5zi9ns324mn80-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/phssz9lgny3mwx26w3z5zi9ns324mn80-gnugrep-3.11/bin/grep -E
checking for fgrep... /nix/store/phssz9lgny3mwx26w3z5zi9ns324mn80-gnugrep-3.11/bin/grep -F
checking for ld used by clang... ld
checking if the linker (ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 786432
checking how to convert aarch64-apple-darwin24.4.0 file names to aarch64-apple-darwin24.4.0 format... func_convert_file_noop
checking how to convert aarch64-apple-darwin24.4.0 file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ranlib... ranlib
checking for archiver @FILE support... @
checking for strip... strip
checking command to parse nm output from clang object... ok
checking for sysroot... no
checking for a working dd... /nix/store/7snvxb9g7cb42vrma4almw8f5wh409rz-coreutils-9.7/bin/dd
checking how to truncate binary pipes... /nix/store/7snvxb9g7cb42vrma4almw8f5wh409rz-coreutils-9.7/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... no
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -no_fixup_chains linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if clang supports -fno-rtti -fno-exceptions... yes
checking for clang option to produce PIC... -fno-common -DPIC
checking if clang PIC flag -fno-common -DPIC works... yes
checking if clang static flag -static works... no
checking if clang supports -c -o file.o... yes
checking if clang supports -c -o file.o... (cached) yes
checking whether the clang linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin24.4.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking for gnutls_hash_init in -lgnutls... no
checking for gcry_control in -lgcrypt... no
checking for gethostbyname in -lws2_32... no
checking for sys/select.h... yes
checking for winsock.h... no
checking for winsock2.h... no
checking for ws2tcpip.h... no
checking for the type of the fifth argument of select()... struct timeval
checking for tzset... yes
checking for sin_len in sock... yes
checking for sys filio.h... yes
checking for arpa/inet.h... yes
checking for netinet/in.h... yes
checking for netinet/tcp.h... yes
checking for poll.h... yes
checking for sys/socket.h... yes
checking for sys/time.h... yes
checking for sys/uio.h... yes
checking for sockaddr_in6 support... yes
checking for SG_IO support... no
checking for iSER support... no
checking for stdatomic.h... yes
checking for pthread support... yes
checking whether libcunit is available... no
configure: You need libcunit to build the test suite.
configure: The scsi/iscsi test suite will not be built.
checking for struct timezone.tz_dsttime... yes
checking for struct CU_SuiteInfo.pSetUpFunc... no
checking for library containing clock_gettime... none required
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating doc/Makefile
config.status: creating examples/Makefile
config.status: creating lib/Makefile
config.status: creating test-tool/Makefile
config.status: creating tests/Makefile
config.status: creating utils/Makefile
config.status: creating libiscsi.pc
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
Running phase: buildPhase
build flags: SHELL=/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash
make  all-recursive
make[1]: входимо до каталогу «/private/tmp/nix-build-libiscsi-1.20.2.drv-0/source»
Making all in lib
make[2]: входимо до каталогу «/private/tmp/nix-build-libiscsi-1.20.2.drv-0/source/lib»
/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I./../include -I./include  -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c -o libiscsipriv_la-connect.lo `test -f 'connect.c' || echo './'`connect.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./../include -I./include -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c connect.c  -fno-common -DPIC -o .libs/libiscsipriv_la-connect.o
/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I./../include -I./include  -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c -o libiscsipriv_la-crc32c.lo `test -f 'crc32c.c' || echo './'`crc32c.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./../include -I./include -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c crc32c.c  -fno-common -DPIC -o .libs/libiscsipriv_la-crc32c.o
/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I./../include -I./include  -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c -o libiscsipriv_la-discovery.lo `test -f 'discovery.c' || echo './'`discovery.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./../include -I./include -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c discovery.c  -fno-common -DPIC -o .libs/libiscsipriv_la-discovery.o
/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I./../include -I./include  -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c -o libiscsipriv_la-init.lo `test -f 'init.c' || echo './'`init.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./../include -I./include -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c init.c  -fno-common -DPIC -o .libs/libiscsipriv_la-init.o
/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I./../include -I./include  -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c -o libiscsipriv_la-login.lo `test -f 'login.c' || echo './'`login.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./../include -I./include -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c login.c  -fno-common -DPIC -o .libs/libiscsipriv_la-login.o
/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I./../include -I./include  -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c -o libiscsipriv_la-nop.lo `test -f 'nop.c' || echo './'`nop.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./../include -I./include -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c nop.c  -fno-common -DPIC -o .libs/libiscsipriv_la-nop.o
/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I./../include -I./include  -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c -o libiscsipriv_la-pdu.lo `test -f 'pdu.c' || echo './'`pdu.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./../include -I./include -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c pdu.c  -fno-common -DPIC -o .libs/libiscsipriv_la-pdu.o
/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I./../include -I./include  -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c -o libiscsipriv_la-iscsi-command.lo `test -f 'iscsi-command.c' || echo './'`iscsi-command.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./../include -I./include -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c iscsi-command.c  -fno-common -DPIC -o .libs/libiscsipriv_la-iscsi-command.o
/nix/store/5mblgwjbmax2k1gksk5mpzak54m2k7z0-bash-5.2p37/bin/bash ../libtool  --tag=CC   --mode=compile clang -DHAVE_CONFIG_H -I. -I..  -I./../include -I./include  -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c -o libiscsipriv_la-multithreading.lo `test -f 'multithreading.c' || echo './'`multithreading.c
libtool: compile:  clang -DHAVE_CONFIG_H -I. -I.. -I./../include -I./include -Wall -W -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-align -Wcast-qual -Wvla -Wno-unknown-warning-option -Wno-stringop-truncation -Wno-unused-parameter -Werror -Wwrite-strings -g -O2 -c multithreading.c  -fno-common -DPIC -o .libs/libiscsipriv_la-multithreading.o
multithreading.c:155:21: error: 'syscall' is deprecated: first deprecated in macOS 10.12 - syscall(2) is unsupported; please switch to a supported interface. For SYS_kdebug_trace use kdebug_signpost(). [-Werror,-Wdeprecated-declarations]
  155 |         pid_t tid = syscall(SYS_gettid);
      |                     ^
/nix/store/nkyq65sn7dlsy6qwym83zwc5w6kqf0y1-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/unistd.h:742:6: note: 'syscall' has been explicitly marked deprecated here
  742 | int      syscall(int, ...);
      |          ^
multithreading.c:257:16: error: 'sem_init' is deprecated [-Werror,-Wdeprecated-declarations]
  257 |         return sem_init(sem, 0, value);
      |                ^
/nix/store/nkyq65sn7dlsy6qwym83zwc5w6kqf0y1-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/semaphore.h:55:42: note: 'sem_init' has been explicitly marked deprecated here
   55 | int sem_init(sem_t *, int, unsigned int) __deprecated;
      |                                          ^
/nix/store/nkyq65sn7dlsy6qwym83zwc5w6kqf0y1-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:196:40: note: expanded from macro '__deprecated'
  196 | #define __deprecated    __attribute__((__deprecated__))
      |                                        ^
multithreading.c:262:16: error: 'sem_destroy' is deprecated [-Werror,-Wdeprecated-declarations]
  262 |         return sem_destroy(sem);
      |                ^
/nix/store/nkyq65sn7dlsy6qwym83zwc5w6kqf0y1-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/semaphore.h:53:26: note: 'sem_destroy' has been explicitly marked deprecated here
   53 | int sem_destroy(sem_t *) __deprecated;
      |                          ^
/nix/store/nkyq65sn7dlsy6qwym83zwc5w6kqf0y1-apple-sdk-11.3/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:196:40: note: expanded from macro '__deprecated'
  196 | #define __deprecated    __attribute__((__deprecated__))
      |                                        ^
3 errors generated.
```

1.20.0 compiles just fine

--- a/configure.ac
+++ b/configure.ac
@@ -242,6 +242,9 @@ fi
 AC_SUBST(LIBS_PRIVATE)
 AM_CONDITIONAL([HAVE_PTHREAD_SPIN_LOCKS], [test x$ac_cv_lib_pthread_pthread_spin_lock = xyes])
 
+# check for pthread_threadid_np
+AC_CHECK_FUNCS(pthread_threadid_np)
+
 AC_CACHE_CHECK([whether libcunit is available],
                [ac_cv_have_cunit],
                [ac_save_CFLAGS="$CFLAGS"

